### PR TITLE
Allow taxon_id_reference for every taxon + add non-required taxon_rank

### PR DESCRIPTION
**Pull request by peterdesmet**

### CODE REVIEW

File: camtrap-dp-profile.json [hunk 1]
```diff
@@ -223,17 +222,38 @@
             "type": "object",
             "required": [
               "taxon_id",
+              "taxon_id_reference",
               "scientific_name"
             ],
             "properties": {
               "taxon_id": {
```
Comment (kbubnicki) [inline camtrap-dp-profile.json:229]: Maybe we should also add `example` ?

File: camtrap-dp-profile.json [hunk 2]
```diff
@@ -223,17 +222,38 @@
...
                 "description": "Unique identifier of the taxon according to the taxonomic reference list defined by `taxon_id_reference`.",
                 "type": "string"
               },
+              "taxon_id_reference": {
+                "description": "URL of the source (reference list) of the `taxon_id`.",
+                "example": "https://www.catalogueoflife.org",
+                "type": "string",
+                "format": "uri"
+              },
               "scientific_name": {
```
Comment (kbubnicki) [inline camtrap-dp-profile.json:239]: Maybe we should also add example ?

**Review by kbubnicki — COMMENTED** [review]
Looks good, just see my comments

File: camtrap-dp-profile.json (hunk 1 above)
Comment (peterdesmet) [reply camtrap-dp-profile.json:229]: Added, see a6026e8

**Comment by peterdesmet** [comment]
@kbubnicki given that `taxon_id` occurs before `scientific_name` in taxonomic (scope), I'm tempted to organize it the same way in `observations`. So `taxon_id, scientific_name` rather than the current `scientific_name, taxon_id`. Do you agree?

**Pull request merged**